app/controllers/mixins/middleware_common_mixin.rb

```diff
@@ -25,4 +29,35 @@ def clear_topology_breadcrumb
       @breadcrumbs.clear
     end
   end
+
+  def init_show(model_class = self.class.model)
+    clear_topology_breadcrumb
+    @lastaction = 'show'
+    @showtype = 'main'
+    @gtl_url = '/show'
+    @display = params[:display] || 'main' unless control_selected?
```

Comment:
Sure not the intent of this PR. BUT: if we have several controller that look very same. If we refactor each in a different way we may not end with more readable code.